### PR TITLE
Add Troubleshooting section;

### PR DESCRIPTION
**Pull request by ShengHuaWu**
The troubleshooting section contains:
When `git ls-remote https://github.com/<your-repo-path>` is executed, it requires username and password. However, If GitHub's two factor authentication is enabled, the process will fail even though both of the username and password are correct. The solution is [generating a Personal Access Token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) and using it as a password.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,3 +45,6 @@ These guides are not to blindly followed but to discuss when in doubt.
 We do test all our apps to ensure that our behavior is what we expect. Initially the codebase was written by `XCTest` framework. Recently we migrated to [Quick](https://github.com/Quick/Quick/) and [Specta](https://github.com/specta/specta).
 
 For the UI tests we use extensively [FBSnapshotTests](https://www.objc.io/issues/15-testing/snapshot-testing/). We use iOS 9.3 as a main version for all UITests and all screenshots are captured in iPhone 6, iPhone 6 Plus, iPhone 5s and iPad Air simulators.
+
+## Troubleshooting
+* When `git ls-remote https://github.com/<your-repo-path>` is executed, it requires username and password. However, If GitHub's two factor authentication is enabled, the process will fail even though both of the username and password are correct. The solution is [generating a Personal Access Token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) and using it as a password.
```
Comment (Mourad-Aly) [inline README.md]: > When `git ls-remote https://github.com/<your-repo-path>` is executed

Did you mean not only `git ls-remote https://github.com/<your-repo-path>` but any git command to any of our private repositories?

> If GitHub's two factor authentication is enabled

Can you clarify what are the two factor authentication?

> The solution is [generating a Personal Access Token]

Can you clarify the solution, after you generate access token, .. ?

Thanks.

Comment (ShengHuaWu) [reply README.md]: @Mourad-Aly Thanks for your comment.

> Did you mean not only git ls-remote https://github.com/<your-repo-path> but any git command to any of our private repositories?

Yes, in fact `git clone`, `git fetch`, `git pull`, and `git push` are all affected.
However, I encountered this issue when executing `make bootstrap` for each project, and the action causes this problem is `git ls-remote https://github.com/<your-repo-path>`. 

> Can you clarify what are the two factor authentication?

Two-factor authentication, or 2FA, is a way of logging into websites that requires more than just a password. [Here](https://help.github.com/articles/about-two-factor-authentication/) is the reference.

> Can you clarify the solution, after you generate access token, .. ?

Once the token is created, use the token as the password. [Here](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/#using-a-token-on-the-command-line) is the reference.

Comment (Mourad-Aly) [reply README.md]: Thanks @ShengHuaWu for your answer.
Can you also include `Once the token is created, use the token as the password. In other words, copy the token from GitHub and paste it into Terminal. Here is the reference.` in your changes?

Comment (ShengHuaWu) [reply README.md]: The sentence has been included in `README`.
Please kindly review.

Comment (Mourad-Aly) [reply README.md]: Perfect, thanks.
Later-on, would be great if you can mention the commit link to the reviewer.
For example you can say:
Included in 18327a5

**Pull request merged**